File: solutions/swift/lasagna/1/Sources/Lasagna/Lasagna.swift
// The 'expectedMinutesInOven' constant
let expectedMinutesInOven: Int = 40

// The 'remainingMinutesInOven' function
func remainingMinutesInOven(elapsedMinutes: Int) -> Int {
    return expectedMinutesInOven - elapsedMinutes
}

// The 'preparationTimeInMinutes' function
func preparationTimeInMinutes(layers: Int) -> Int {
    return 2 * layers
}

// The 'totalTimeInMinutes' function
func totalTimeInMinutes(layers: Int, elapsedMinutes: Int) -> Int {
    return preparationTimeInMinutes(layers: layers) + elapsedMinutes
}

File: solutions/java/lasagna/2/src/main/java/Lasagna.java
public class Lasagna {
    private static final int MINUTES_IN_OVEN = 40;
    private static final int MINUTES_PER_LAYER = 2;
    
    public int expectedMinutesInOven() {
        return MINUTES_IN_OVEN;
    }

    public int remainingMinutesInOven(int actualMinutes) {
        return expectedMinutesInOven() - actualMinutes;
    }

    public int preparationTimeInMinutes(int layers) {
        return layers * MINUTES_PER_LAYER;
    }

    public int totalTimeInMinutes(int layers, int minutesInOven) {
        return preparationTimeInMinutes(layers) + minutesInOven;
    }
}
